docs/postgresql.md

```diff
@@ -0,0 +1,139 @@
+# `postgresql` Metrics Receiver
+
+The postgresql receiver can retrieve stats from your postgresql instance by connecting as a monitoring user.
+
+## Prerequisites
+
+The `postgresql` receiver defaults to connecting to a local postgresql server using a Unix socket and Unix authentication as the `postgres` user.
+
+## Configuration
+
+Following the guide for [Configuring the Ops Agent](https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#file-location), add the required elements for your postgresql configuration.
+
+To configure a receiver for your postgresql metrics, specify the following fields:
+
+| Field                   | Default                         | Description |
+| ---                     | ---                             | ---         |
+| `type`                  | required                        | Must be `postgresql`. |
+| `endpoint`              | `/var/run/postgresql/.s.PGSQL.5432`   | The hostname:port or socket path used to connect to postgresql |
+| `collection_interval`   | `60s`                           | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
+| `username`              | `postgres`                          | The username used to connect to the server. |
```

Comment:
Can we ensure invalid config combinations throw errors as expected?

e.g. username present but password not or vice versa, or if any combo in the TLS configs are not valid.

Reply:
I don't see a way to use the validator library we're using to do something like "if insecure=true, do not allow this field" for ca_file/cert_file/key_file.

Password is not always required even when username is provided, because of the Ident/OS authentication mode of auth.